AsyncDisplayKit/Debug/ASLayoutElementInspectorNode.m

```diff
@@ -0,0 +1,507 @@
+  
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    __inspector = [[ASLayoutElementInspectorNode alloc] init];
+  });
+  
+  return __inspector;
+}
+
+#pragma mark - lifecycle
+- (instancetype)init
+{
+  self = [super init];
+  if (self) {
+    
+    _tableNode                 = [[ASTableNode alloc] init];
+    _tableNode.delegate        = self;
+    _tableNode.dataSource      = self;
+
+    [self addSubnode:_tableNode];  // required because of manual layout
+  }
+  return self;
+}
+
+- (void)didLoad
+{
+  [super didLoad];
+  _tableNode.view.backgroundColor = [UIColor colorWithRed:40/255.0 green:43/255.0 blue:53/255.0 alpha:1];
+  _tableNode.view.separatorStyle = UITableViewCellSeparatorStyleNone;
+  _tableNode.view.allowsSelection = NO;
+  _tableNode.view.sectionHeaderHeight = 40;
+}
+
+- (void)layout
+{
+  [super layout];
+  _tableNode.frame = self.bounds;
+}
+
+#pragma mark - intstance methods
+- (void)setLayoutElementToEdit:(id<ASLayoutElement>)layoutElementToEdit
+{
+  if (_layoutElementToEdit != layoutElementToEdit) {
+    _layoutElementToEdit = layoutElementToEdit;
+  }
+  [_tableNode reloadData];
+}
+
+#pragma mark - ASTableDataSource
+
+- (ASCellNode *)tableView:(ASTableView *)tableView nodeForRowAtIndexPath:(NSIndexPath *)indexPath
+{
+  if (indexPath.section == 0) {
+    
+    NSDictionary *attributes = @{NSForegroundColorAttributeName : [UIColor colorWithRed:255/255.0 green:181/255.0 blue:68/255.0 alpha:1],
+                                 NSFontAttributeName : [UIFont fontWithName:@"Menlo-Regular" size:12]};
+    ASTextCellNode *textCell = [[ASTextCellNode alloc] initWithAttributes:attributes insets:UIEdgeInsetsMake(0, 4, 0, 0)];
+    textCell.text = [_layoutElementToEdit description];
+    return textCell;
+    
+  } else {
+    
+    return [[ASLayoutElementInspectorCell alloc] initWithProperty:(ASLayoutElementPropertyType)indexPath.row layoutElementToEdit:_layoutElementToEdit];
+  }
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
+{
+  if (section == 0) {
+    return 1;
+  } else {
+    return ASLayoutElementPropertyCount;
+  }
+}
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
+{
+  return 2;
+}
+
+- (nullable UIView *)tableView:(UITableView *)tableView viewForHeaderInSection:(NSInteger)section
+{
+  UILabel *headerTitle = [[UILabel alloc] initWithFrame:CGRectZero];
+  
+  NSString *title;
+  if (section == 0) {
+    title = @"<Layoutable> Item";
+  } else {
+    title = @"<Layoutable> Properties";
+  }
+  
+  NSDictionary *attributes = @{NSForegroundColorAttributeName : [UIColor whiteColor],
+                               NSFontAttributeName : [UIFont fontWithName:@"Menlo-Bold" size:12]};
+  headerTitle.attributedText = [[NSAttributedString alloc] initWithString:title attributes:attributes];
+  
+  return headerTitle;
+}
+
+
+//- (ASLayoutSpec *)layoutSpecThatFits:(ASSizeRange)constrainedSize
```

Comment:
This particular file has a bunch of commented-out code, but I'd like to leave it in place for this initial commit.  Like many of our +Beta or +Debug headers, it's still a work in progress.

The most important thing to its success in becoming an official feature is allowing time to focus on improvements rather than the continued complex rebasing it has required.

Fortunately, the value of an Inspector like this isn't speculative - we can be quite confident that this will be one of our most powerful debugging (and educational) features once it is finished.